tests/test_wrappers.py

```diff
@@ -1199,3 +1199,137 @@ def test_pprint(self):
+
+class TestPartialFit:
+    @pytest.mark.parametrize(
+        "config",
+        ["MLPRegressor", "MLPClassifier", "CNNClassifier", "CNNClassifierF"],
+    )
+    def test_partial_fit(self, config):
+        loader, model, build_fn, _ = CONFIG[config]
+        clf = model(build_fn, epochs=1)
+        data = loader()
+
+        X, y = data.data[:100], data.target[:100]
+        clf.partial_fit(X, y)
+
+        assert len(clf.history_["loss"]) == 1
+        clf.partial_fit(X, y)
+        assert len(clf.history_["loss"]) == 2
+
+        # Make sure new model not created
+        model = clf.model_
+        clf.partial_fit(X, y)
+        assert (
+            clf.model_ is model
+        ), "Model memory address should remain constant"
+
+    def test_partial_fit_history_len(self, config="CNNClassifier"):
+        # history_ records the history from this partial_fit call
+        # Make sure for each call to partial_fit a single entry
+        # into the history is added
+        # As per https://github.com/keras-team/keras/issues/1766,
+        # there is no direct measure of epochs
+        loader, model, build_fn, _ = CONFIG[config]
+        clf = model(build_fn, epochs=1)
+        data = loader()
+
+        X, y = data.data[:100], data.target[:100]
+        for k in range(10):
+            clf = clf.partial_fit(X, y)
+            assert len(clf.history_["loss"]) == k + 1
+            assert set(clf.history_.keys()) == {"loss", "accuracy"}
+
+    @pytest.mark.parametrize(
+        "config", ["CNNClassifier", "CNNClassifierF"],
+    )
+    def test_pf_pickle_pf(self, config):
+        loader, model, build_fn, _ = CONFIG[config]
+        clf = model(build_fn, epochs=1)
+        data = loader()
+
+        X, y = data.data[:100], data.target[:100]
+        clf.partial_fit(X, y)
+
+        # Check that partial_fit -> pickle -> partial_fit
+        # builds up the training
+        # even after pickling by checking that
+        # (1) the history_ attribute grows in length
+        # (2) the loss value decreases
+        clf2 = clf
+        for k in range(2, 4):
+            clf2 = pickle.loads(pickle.dumps(clf2))
+            clf2.partial_fit(X, y)
+            assert len(clf.history_["loss"]) == 1
+            assert len(clf2.history_["loss"]) == k
+            assert np.allclose(
+                clf.history_["loss"][0], clf2.history_["loss"][0]
+            )
+
+        weights1 = [w.numpy() for w in clf.model_.weights]
+        weights2 = [w.numpy() for w in clf2.model_.weights]
+        n_weights = [w1.size for w1 in weights1]
+        assert 1000 <= sum(n_weights) <= 2000
+        assert 200 <= np.mean(n_weights) <= 300
+        assert max(n_weights) >= 1000
+        rel_errors = [
+            np.linalg.norm(w1 - w2) / np.linalg.norm((w1 + w2) / 2)
+            for w1, w2 in zip(weights1, weights2)
+        ]
+        assert len(rel_errors) == 4
+        assert any(x > 0.5 for x in rel_errors)
+        assert all(0.01 < x for x in rel_errors)
```

Comment:
This looks great! Could you add some comments to clarify what the constants are and such?

Reply:
Added.